skylib/zip.bzl

```diff
@@ -14,36 +14,65 @@
 """Functions for producing the gzip of an artifact."""
 
 def gzip(ctx, artifact):
-    """Create an action to compute the gzipped artifact."""
+    """Create an action to compute the gzipped artifact.
+
+    Args:
+       ctx: The context
+       artifact: The artifact to zip
+
+    Returns:
+       the gzipped artifact.
+    """
     out = ctx.actions.declare_file(artifact.basename + ".gz")
+    toolchain_info = ctx.toolchains["@io_bazel_rules_docker//toolchains/docker:toolchain_type"].info
+    input_manifests = []
+    tools = []
+    gzip_path = toolchain_info.gzip_path
+    if toolchain_info.gzip_target:
+        gzip_path = toolchain_info.gzip_target.files.to_list()[0].path
+        tools, _, input_manifests = ctx.resolve_command(tools = [toolchain_info.gzip_target])
+    elif toolchain_info.gzip_path == "":
+        fail("gzip could not be found. Make sure it is in the path or set it " +
+             "explicitly in the docker_toolchain_configure")
+
     ctx.actions.run_shell(
-        command = "%s -n < %s > %s" % (ctx.executable.gzip.path, artifact.path, out.path),
+        command = "%s -n < %s > %s" % (gzip_path, artifact.path, out.path),
+        input_manifests = input_manifests,
         inputs = [artifact],
         outputs = [out],
         use_default_shell_env = True,
         mnemonic = "GZIP",
-        tools = [ctx.executable.gzip],
+        tools = tools,
     )
     return out
 
 def gunzip(ctx, artifact):
-    """Create an action to compute the gunzipped artifact."""
+    """Create an action to compute the gunzipped artifact.
+
+    Args:
+       ctx: The context
+       artifact: The artifact to zip
+
+    Returns:
+       the gunzipped artifact.
+    """
     out = ctx.actions.declare_file(artifact.basename + ".nogz")
+    toolchain_info = ctx.toolchains["@io_bazel_rules_docker//toolchains/docker:toolchain_type"].info
+    input_manifests = []
+    tools = []
+    gzip_path = toolchain_info.gzip_path
+    if toolchain_info.gzip_target:
+        gzip_path = toolchain_info.gzip_target.files.to_list()[0].path
```

Comment:
maybe `toolchain_info.gzip_target.files_to_run.executable.path`?

Reply:
thanks, that seems to work. I also folded your example as a test as part of this PR. While I cant test it on remote execution we can have a test that verifies gzip as tool works as intended. Thanks for your help!